java/maven/Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM maven:3.5.0-jdk-8
+
+ADD poms /poms
+
+# for each included pom file, resolve all of its dependencies
+RUN for POM in $(find /poms/*.xml); do \
+  mvn --batch-mode -f $POM -s /usr/share/maven/ref/settings-docker.xml org.apache.maven.plugins:maven-dependency-plugin:3.0.0:go-offline -DincludeParents ; \
+  done; \
+  rm -r /poms
```

Comment:
this won't reduce the size of the image, I assume that's ok and we just don't want to leave artifacts around for the customer to see?